### ISSUE TITLE
MapDotKeyReplacer throws an ConcurrentModificationException

### ISSUE THREAD

**Issue by luca010** [javers/javers#910]
MapKeyDotReplacer throws an ConcurrentModificationException when tries to replace dots in the document Map. I would suggest to rewrite the method replaceInMapKeys in the MapKeyDotReplacer class as follows:
```java
private Document replaceInMapKeys(Document map, String regexFrom, String from, String to) {
    Document result = new Document();
    for (String key : map.keySet()) {
        Object val = map.get(key);
        if (key.contains(from)) {
            String escaped = key.replaceAll(regexFrom, to);
            result.put(escaped, val);
        } else {
           result.put(key, val);
        }
    }
    return result;
}
```

**Comment by luca010** [comment]
Hi @bartoszwalacik , I forked this repository and I added a new [test case](https://github.com/luca010/javers/blob/master/javers-persistence-mongo/src/test/groovy/org/javers/repository/mongo/cases/KeyMapDotReplacerTest.groovy) to reproduce the bug. I hope this will help you.

**Comment by bartoszwalacik** [comment]
Great, so now it shuld be easy for you to contribute the fix :)

**Comment by luca010** [comment]
I pushed the fix on my fork. I run all the test cases with success.

**Comment by bartoszwalacik** [comment]
ok, please create the pull request from your fork to javers master

**Comment by bartoszwalacik** [comment]
fix is released in 5.8.5
@luca010 thanks for your contribution, well done
